crypto/property/property.c

```diff
@@ -549,6 +563,41 @@ int ossl_method_store_remove_all_provided(OSSL_METHOD_STORE *store,
     return 1;
 }
 
+int ossl_method_store_freeze(OSSL_METHOD_STORE *store, const char *propq)
+{
+    if (store == NULL || store->frozen_store != NULL)
+        return 0;
+
+    store->frozen_store = OPENSSL_zalloc(sizeof(store->frozen_store));
+    if (store->frozen_store == NULL)
+        return 0;
+    if (propq != NULL) {
+        store->frozen_store->propq = OPENSSL_strdup(propq);
```

Comment:
What will you do with the propq?

Reply:
I've also wrote tests to make sure the freeze works with/without the prop query